src/main/kotlin/com/swisscom/health/des/cdr/client/handler/PullFileHandling.kt

```
@@ -93,15 +101,32 @@ class PullFileHandling(
      * @param connector the connector for whom the file was requested
      * @param file the file to move
      */
-    private fun moveFileToClientFolder(connector: CdrClientConfig.Connector, file: Path): Boolean {
+    private fun moveFileToClientFolder(connector: CdrClientConfig.Connector, file: Path) {
         logger.debug { "Move file to target directory start" }
-        val sourceFile = file
-        val targetFolder = connector.targetFolder
-        val targetTmpFile = targetFolder.resolve(sourceFile.name)
+        val targetFolder = if (connector.typeFolders.isEmpty()) {
+            connector.targetFolder
+        } else {
+            val forumDatenaustauschNamespaces: ForumDatenaustauschNamespaces = try {
+                xmlParser.findSchemaDefinition(file.inputStream().use { it.toDom() })
+            } catch (e: SAXException) {
```

Why not catch all exception types? No matter the exception type, the operation has failed?